# Практическая работа №2: Обработка выборочных данных. Нахождение точечных оценок параметров распределения

Выполнили студенты гр. 2384 Пчелинцева Кристина и Лукьянова Мария.

## Цель работы

Получение практических навыков нахождения точечных статистических оценок параметров распределения.

## Основные теоретические положения

**Статистическая оценка $\Theta^*$** неизвестного параметра теоритического распределения $\Theta$ - фунцкия от наблюдаемых значений случайной величины  $$\Theta^*=f(x_1, ... , x_n)$$

**Точечные оценки** - стат. оценки, определяемые одним числом

**Несмещенная оценка**, если выполнено:

$$\mathbb{E}(\Theta^*) = \Theta$$

**Эффективная оценка**, если выполнено при заданном объеме выборки n:

$$\mathbb{D}(\Theta^*) \to min$$

**Состоятельная оценка для параметра $\Theta^*$**, если:

$$\underset{N \to \inf}{lim }\mathbb{P}(|\Theta - \Theta^*| \le \epsilon)=1,$$

где $\epsilon > 0$ сколь угодо малое число.

**Начальный эмпирический момент k-ого порядка:**

$$\bar{M}_k = \frac{1}{N}\sum n_j x_j^k, $$

где N - объем выборки, $n_j$ - абсолютная частота, $x_j$ - середина j-ого интервала

**Центральный эмпирический момент k-ого порядка:**
$$\bar{m}_k = \frac{1}{N}\sum n_j (x_j^k - \bar{x}_в ), $$

где $\bar{x}_в$ - выборочное среднее, равное $\bar{M}_1$

**Выборочная дисперсия:**

$$\mathbb{D_в} = \bar{m}_2 = \frac{1}{N}\sum n_j(x_j - \bar{x}_в)^2$$

**Исправленная оценка дисперсии:**

$$s^2 = \frac{N}{N-1}\mathbb{D_в}$$

**Статистическая оценка СКО:**

$$\sigma_в = \sqrt{\mathbb{D_в}}$$

**Статистическая оценка асимметрии:**

$$\bar{A}_s = \frac{\bar{m}_3}{s^3}$$

**Статистическая оценка эксцесса:**

$$\bar{E} = \frac{\bar{m}_4}{s^4} - 3$$

**Условная варианта:**

$$u_j = \frac{x_j - C}{h},$$

где $x_j$ - середина интервала, С - условный ноль (середина интервального ряда)

**Условный момент k-ого порядка:**

$$\bar{M}_k^* = \frac{1}{N}\sum n_j u_j^k = \frac{1}{N}\sum n_j\left( \frac{x_j-C}{n} \right)^k$$

робастные выборочные статистические оценки:

1) **Мода:**

$$M_o^* = x_{i-1} + h \frac{\tilde{m}_i - \tilde{m}_{i-1}}{(\tilde{m}_i - \tilde{m}_{i-1})+(\tilde{m}_i - \tilde{m}_{i+1})}, $$

где $x_{i-1}$ - левая граница интервала, содержащего значение моды; h - длина интервалов; \tilde{m}_i - относительная частота i-ого интервала.

2) **Медиана:**
$$M_e^* = x_{i-1} + h \frac{0.5-\sum_{j=1}^{i-1}h\ \tilde{m}_j}{\tilde{m}_i}$$

3) **Коэффициент вариации:**

$$V^* = \frac{\sigma_в}{\bar{x}_в}\cdot 100%$$











## Постановка задачи

Для заданных выборочных данных вычислить с использованием метода моментов и условных вариант точечные статистические оценки математического ожидания, дисперсии, среднеквадратичного отклонения, асимметрии, эксцесса, моды, медианы и коэффициента вариации исследуемой случайной величины. Полученные результаты содержательно проинтерпретировать.

## Выполнение работы

### Пункт 1.
Для середин интервального ряда, полученного в практической работе №1, вычислить условные варианты. Заполнить табл. 1 (в последней строке Σ необходимо заполнить суммы столбцов; ячейки отмеченные прочерком заполнять не надо). Провести контроль вычислений.

In [451]:
import pandas as pd
import numpy as np

In [452]:
x = np.array([88.73, 96.66, 104.5585, 112.457, 120.3555, 128.254, 136.1525, 144.051]) 
n = np.array([0.009009, 0.180180, 0.099099, 0.261261, 0.279279, 0.000000, 0.036036, 0.135135]) 
m = np.array([1, 20, 11, 29, 31, 0, 4, 15]) 
h = 7.90 
intervals = np.array(
    [[84.749, 92.711], 
     [92.711, 100.609], 
     [100.609, 108.508], 
     [108.508, 116.406], 
     [116.406, 124.305], 
     [124.305, 132.203], 
     [132.203, 140.102], 
     [140.102, 148.0]])

C - условный ноль, равный значению варианты среднего значения интервального ряда.

In [453]:
C = x[len(x) // 2]
u = np.round((x - C) / h).astype(int)
print(u)

[-4 -3 -2 -1  0  1  2  3]


In [454]:
nu = n*u
nu2 = n*u**2
nu3 = n*u**3
nu4 = n*u**4
nup14 =n*(u+1)**4

Таблица 1

| i | $x_i$ | $n_i$ | $u_i$ | $n_iu_i$ | $n_iu_i^2$ | $n_iu_i^3$ | $n_iu_i^4$ | $n_i(u_i-1)^4$ |
|---|---|---|---|---|---|---|---|---|
| 1 | 88.73 | 0.009009 | -4 | -0.0360 | 0.1441 | -0.5765 | 2.3063 | 0.7297 |
| 2 | 96.66 | 0.180180 | -3 | -0.5405 | 1.6216 | -4.8648 | 14.5945 | 2.8828 |
| 3 | 104.5585 | 0.099099 | -2 | -0.1981 | 0.3963 | -0.7927 | 1.5855 | 0.0991 |
| 4 | 112.457 | 0.261261 | -1 | -0.2612 | 0.2612 | -0.2612 | 0.2612 | 0.0 |
| 5 | 120.3555 | 0.279279 | 0 | 0.0 | 0.0 | 0.0 | 0.0 | 0.2792 |
| 6 | 128.254 | 0.0 | 1 | 0.0 | 0.0 | 0.0 | 0.0 | 0.0 |
| 7 | 136.1525 | 0.036036 | 2 | 0.0721 | 0.1441 |0.2882 | 0.5765 | 2.9189 |
| 8 | 144.051 | 0.135135 | 3 | 0.4054 | 1.2162 | 3.6486 | 10.9459 | 34.5945 |
|  |$\sum$| 1.0 | - | -0.5585 | 3.7837 | -2.5585 | 30.2702 | 41.5044 |

In [455]:
# контроль вычислений
if(round(sum(nup14),6) == round((sum(nu4) + 4*sum(nu3) + 6*sum(nu2) + 4*sum(nu) + sum(n)),6)):
  print("Вычисления верны!")
else:
  print("Вычисления ошибочны!")

Вычисления верны!


### Пункт 2.
Вычислить условные эмпирические моменты $\bar{M}^*_k$ через условные варианты. С помощью условных эмпирических моментов вычислить центральные эмпирические моменты $\bar{m}_k$. Полученные результаты занести в табл. 2.

In [456]:
M = [sum(nu), sum(nu2), sum(nu3), sum(nu4)]
x_mean = M[0]*h + C 
m_central = [
    0,  
    (M[1] - M[0]**2) * h**2,
    (M[2] - 3*M[1]*M[0] + 2*M[0]**3) * h**3,
    (M[3] - 4*M[2]*M[0] + 6*M[1]*M[0]**2 - 3*M[0]**4) * h**4
]

# print("| k | M_k* | m_k |")
# print(f"| 1 | {M[0]:.4f} | {m_central[0]:.4f} |")  
# print(f"| 2 | {M[1]:.4f} | {m_central[1]:.4f} |")
# print(f"| 3 | {M[2]:.4f} | {m_central[2]:.4f} |")
# print(f"| 4 | {M[3]:.4f} | {m_central[3]:.4f} |")

Таблица 2

| k | $\bar{M}^*_k$ | $\bar{m}_k$ |
|---|---|---|
| 1 | -0.5586 | 0.0000 |
| 2 | 3.7838 | 216.6746 |
| 3 | -2.5586 | 1692.7508 |
| 4 | 30.2702 | 122088.1194 |

### Пункт 3.
Вычислить выборочные среднее $\bar x_{\textit{в}}$ и дисперсию $D_{\textit{в}}$ с помощью стандартной формулы и с помощью условных вариант. Убедиться, что результаты совпадают. Вычислить выборочное СКО $\sigma_{\textit{в}}$.



In [457]:
# вычисление выборочного среднего и дисперсии (стандартная формула)
x_mean_std = sum(n*x)  
variance_std = sum(n*(x - x_mean_std)**2)
print("Выборочное среднее (стандарт):", round(x_mean_std, 4))
print("Выборочная дисперсия (стандарт):", round(variance_std, 4))

# вычисление выборочного среднего и дисперсии (условные варианты)
x_mean_cond = M[0]*h + C
variance_cond = m_central[1]  

print("Выборочное среднее (условн.):", round(x_mean_cond, 4))
print("Выборочная дисперсия (условн.):", round(variance_cond, 4))

# проверка
if (round(x_mean_std, 2) == round(x_mean_cond, 2) and 
    round(variance_std, 2) == round(variance_cond, 2)):
    print("Значения совпадают!")
else:
    print("Значения различаются!")
    print(f"Разница в среднем: {abs(x_mean_std - x_mean_cond):.4f}")
    print(f"Разница в дисперсии: {abs(variance_std - variance_cond):.4f}")

standard_deviation = np.sqrt(variance_std)
print("Выборочное СКО:", round(standard_deviation, 4))

Выборочное среднее (стандарт): 115.9433
Выборочная дисперсия (стандарт): 216.6077
Выборочное среднее (условн.): 115.9429
Выборочная дисперсия (условн.): 216.6746
Значения различаются!
Разница в среднем: 0.0004
Разница в дисперсии: 0.0669
Выборочное СКО: 14.7176


### Пункт 4.

Вычислить исправленную выборочную дисперсию $s^2$ и исправленное СКО $s$. Сравнить данные оценки со смещёнными оценками дисперсии и СКО. Сделать выводы

In [458]:
N = 111
s2 = N/(N-1)*variance_cond
print("Исправленная выборочная дисперсия:", round(s2,4))

s = s2**(1/2)
print("Исправленное СКО", round(s,4))


Исправленная выборочная дисперсия: 218.6444
Исправленное СКО 14.7866


Исправленные оценки немного больше смещенных оценок. Они отличаются из-за смещения, которое образуется при расчете с использованием статистик, которые основываются на данных выборки.

### Пункт 5.

Найти статистическую оценку коэффициентов асимметрии $\bar{A}_s$ и эксцесса $\bar{E}$. Сделать выводы.


In [459]:
A_s = m_central[2] / s**3  
E = m_central[3] / s**4 - 3

print("Статистическая оценка асимметрии:", round(A_s,4))
print("Статистическая оценка эксцесса:", round(E,4))


Статистическая оценка асимметрии: 0.5236
Статистическая оценка эксцесса: -0.4461


Статистическая оценка коэффициента асимметрии свидетельствует о наличии положительной (правосторонней) асимметрии исследуемого распределения. Это означает, что большая часть вариант сосредоточена левее среднего значения, а правая часть хвоста распределения является более протяженной.

Отрицательное значение эксцесса характеризует распределение как платикуртическое, то есть более плосковершинное по сравнению с нормальным законом распределения. Данный факт указывает на относительно легкие хвосты распределения и меньшую вероятность появления резко выделяющихся значений (выбросов).

### Пункт 6.

Для интервального ряда вычислить моду $M^*_o$, медиану $M^*_e$ и коэффициент вариации $V^*$ заданного распределения. Сделать выводы.

In [460]:
max_index = np.argmax(n)

x0 = intervals[max_index][0]

f_mo = n[max_index]  
f_prev = n[max_index - 1] if max_index > 0 else 0  # частота предыдущего
f_next = n[max_index + 1] if max_index < len(n) - 1 else 0  # частота следующего

M_o = x0 + h * (f_mo - f_prev) / ((f_mo - f_prev) + (f_mo - f_next))
print("Мода:", round(M_o, 4))

i = 0
cumsum_freq = np.cumsum(n)
for k in range(len(n)):
    if cumsum_freq[k] >= 0.5:
        i = k
        break

M_e = intervals[i][0] + h * (0.5 - cumsum_freq[i-1])/n[i]
print("Медиана", round(M_e,4))

V = standard_deviation/x_mean * 100
print(f"Выборочный коэффициент вариации: {round(V)}%")

Мода: 116.8848
Медиана 114.9097
Выборочный коэффициент вариации: 13%


Мода близка к значению медианы и смещена вправо относительно центрального значения в интервальном ряде. Поэтому можно говорить только о небольшой асимметричности распределения. Выборочный коэффициент вариации равен примерно 13%, что говорит об  умеренном разбросе данных относительно среднего.

Сделаем аналогичные шаги для второй выборки.

In [461]:
x = np.array([1990.243, 2478.229, 2964.271, 3450.3125, 3936.354, 4422.396, 4908.4375, 5394.479])  
n = np.array([0.117117, 0.315315, 0.279279, 0.081081, 0.117117, 0.045045, 0.018018, 0.027027])  
abs_freq = np.array([13, 35, 31, 9, 13, 5, 2, 3])  
h = 486.04  
intervals = np.array([
    [1745.278, 2235.208],
    [2235.208, 2721.25],
    [2721.25, 3207.292],
    [3207.292, 3693.333],
    [3693.333, 4179.375],
    [4179.375, 4665.417],
    [4665.417, 5151.458],
    [5151.458, 5637.5]
])

max_freq_index = np.argmax(abs_freq)  
C = x[max_freq_index]  
print(f"Правильный C = {C}")

u = np.round((x - C) / h).astype(int)
print(f"Правильные u = {u}")  

nu = n*u
nu2 = n*u**2
nu3 = n*u**3
nu4 = n*u**4
nup14 =n*(u+1)**4

if(round(sum(nup14),6) == round((sum(nu4) + 4*sum(nu3) + 6*sum(nu2) + 4*sum(nu) + sum(n)),6)):
  print("Вычисления верны!\n")
else:
  print("Вычисления ошибочны!\n")


df = pd.DataFrame({
    "i": np.arange(1, len(x)+1),
    "x_i": x,
    "n_i": n,
    "u_i": u,
    "n_i u_i": n * u,
    "n_i u_i^2": n * u**2,
    "n_i u_i^3": n * u**3,
    "n_i u_i^4": n * u**4,
    "n_i (u_i+1)^4": n * (u + 1)**4
})

sum_row = df.sum(numeric_only=True)
sum_row["i"] = ""
sum_row["x_i"] = ""
sum_row["u_i"] = "-"

df.loc["Σ"] = sum_row

df

Правильный C = 2478.229
Правильные u = [-1  0  1  2  3  4  5  6]
Вычисления верны!



C:\Users\pchel\AppData\Local\Temp\ipykernel_19960\277084097.py:48: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.
  sum_row["i"] = ""


,i,x_i,n_i,u_i,n_i u_i,n_i u_i^2,n_i u_i^3,n_i u_i^4,n_i (u_i+1)^4
0,1,1990.243,0.117117,-1,-0.117117,0.117117,-0.117117,0.117117,0.000000
1,2,2478.229,0.315315,0,0.000000,0.000000,0.000000,0.000000,0.315315
2,3,2964.271,0.279279,1,0.279279,0.279279,0.279279,0.279279,4.468464
3,4,3450.3125,0.081081,2,0.162162,0.324324,0.648648,1.297296,6.567561
4,5,3936.354,0.117117,3,0.351351,1.054053,3.162159,9.486477,29.981952
5,6,4422.396,0.045045,4,0.180180,0.720720,2.882880,11.531520,28.153125
6,7,4908.4375,0.018018,5,0.090090,0.450450,2.252250,11.261250,23.351328
7,8,5394.479,0.027027,6,0.162162,0.972972,5.837832,35.026992,64.891827
Σ,,,0.999999,-,1.108107,3.918915,14.945931,68.999931,157.729572


In [462]:
M = [sum(nu), sum(nu2), sum(nu3), sum(nu4)]
x_mean = M[0]*h + C 
m_central = [
    0,  
    (M[1] - M[0]**2) * h**2,
    (M[2] - 3*M[1]*M[0] + 2*M[0]**3) * h**3,
    (M[3] - 4*M[2]*M[0] + 6*M[1]*M[0]**2 - 3*M[0]**4) * h**4
]

print("| k | M_k* | m_k |")
print(f"| 1 | {M[0]:.4f} | {m_central[0]:.4f} |")  
print(f"| 2 | {M[1]:.4f} | {m_central[1]:.4f} |")
print(f"| 3 | {M[2]:.4f} | {m_central[2]:.4f} |")
print(f"| 4 | {M[3]:.4f} | {m_central[3]:.4f} |")

# вычисление выборочного среднего и дисперсии (стандартная формула)
x_mean_std = sum(n*x)  
variance_std = sum(n*(x - x_mean_std)**2)
print("Выборочное среднее (стандарт):", round(x_mean_std, 4))
print("Выборочная дисперсия (стандарт):", round(variance_std, 4))

# вычисление выборочного среднего и дисперсии (условные варианты)
x_mean_cond = M[0]*h + C
variance_cond = m_central[1]  

print("Выборочное среднее (условн.):", round(x_mean_cond, 4))
print("Выборочная дисперсия (условн.):", round(variance_cond, 4))

# проверка
if (round(x_mean_std, 2) == round(x_mean_cond, 2) and 
    round(variance_std, 2) == round(variance_cond, 2)):
    print("Значения совпадают!")
else:
    print("Значения различаются!")
    print(f"Разница в среднем: {abs(x_mean_std - x_mean_cond):.6f}")
    print(f"Разница в дисперсии: {abs(variance_std - variance_cond):.6f}")

standard_deviation = np.sqrt(variance_std)
print("Выборочное СКО:", round(standard_deviation, 4))

N = 111
s2 = N/(N-1)*variance_std
print("\nИсправленная выборочная дисперсия:", round(s2,4))

s = s2**(1/2)
print("Исправленное СКО", round(s,4))

A_s = m_central[2] / s**3
E = m_central[3] / s**4 - 3

print("Статистическая оценка асимметрии:", round(A_s,4))
print("Статистическая оценка эксцесса:", round(E,4))

max_index = np.argmax(n)
x0 = intervals[max_index][0]
f_mo = n[max_index]  
f_prev = n[max_index - 1] if max_index > 0 else 0
f_next = n[max_index + 1] if max_index < len(n) - 1 else 0
M_o = x0 + h * (f_mo - f_prev) / ((f_mo - f_prev) + (f_mo - f_next))
print("Мода:", round(M_o, 4))

cumsum_freq = np.cumsum(n)
for k in range(len(n)):
    if cumsum_freq[k] >= 0.5:
        i = k
        break
M_e = intervals[i][0] + h * (0.5 - cumsum_freq[i-1])/n[i]
print("Медиана", round(M_e,4))

V = standard_deviation / x_mean_std * 100
print(f"Выборочный коэффициент вариации: {round(V)}%")

| k | M_k* | m_k |
| 1 | 1.1081 | 0.0000 |
| 2 | 3.9189 | 635711.3445 |
| 3 | 14.9459 | 532704535.6779 |
| 4 | 68.9999 | 1512489463556.8645 |
Выборочное среднее (стандарт): 3016.5851
Выборочная дисперсия (стандарт): 636182.5062
Выборочное среднее (условн.): 3016.8133
Выборочная дисперсия (условн.): 635711.3445
Значения различаются!
Разница в среднем: 0.228221
Разница в дисперсии: 471.161653
Выборочное СКО: 797.6105

Исправленная выборочная дисперсия: 641965.9835
Исправленное СКО 801.2278
Статистическая оценка асимметрии: 1.0357
Статистическая оценка эксцесса: 0.67
Мода: 2646.4726
Медиана 2838.8412
Выборочный коэффициент вариации: 26%


<!-- | k | $\bar{M}^*_k$ | $\bar{m}_k$ |
|---|---|---|
| 1 | -1.8919 | 3016.8198 |
| 2 | 6.2703 | 635711.9 |
| 3 | -17.4054 | 532702688.5835 |
| 4 | 62.5945 | 1512486987376.3823 | -->

В этой выборке исправленные оценки (801.23) незначительно отличаются от смещенных (797.61), что связано с большим объемом выборки (N=111). Оценка асимметрии положительная (1.0357), что указывает на правую асимметрию — хвост распределения вытянут вправо, при этом мода (2646.47) меньше медианы (2838.84) и среднего (3016.59). Эксцесс небольшой положительный (0.67), что свидетельствует о лептокуртическом распределении (островершинном) с тяжелыми хвостами. Выборочный коэффициент вариации равен примерно 26%, что говорит об умеренном разбросе данных относительно среднего, но совокупность остается однородной.

## Вывод

В ходе выполнения практической работы были успешно достигнуты поставленные цели, связанные с вычислением точечных статистических оценок параметров распределения. Для двух интервальных рядов проведен анализ данных, включая расчет математического ожидания, дисперсии, среднеквадратичного отклонения, асимметрии, эксцесса, моды, медианы и коэффициента вариации. Оба метода расчёта — стандартный и через условные варианты — подтвердили корректность результатов, несмотря на небольшие расхождения, вызванные погрешностью округления относительных частот.

Для первой выборки выявлена умеренная положительная асимметрия (0.5238), указывающая на правосторонний хвост распределения, где среднее значение (115.94) незначительно превышает медиану (114.91) и моду (116.88). Отрицательный эксцесс (-0.4446) отражает платикуртический характер распределения с плоской вершиной и лёгкими хвостами. Коэффициент вариации (13%) свидетельствует об однородности совокупности и умеренном разбросе данных относительно среднего.

Во второй выборке обнаружена значительная положительная асимметрия (1.0357), характерная для правостороннего смещения, где мода (2646.47) меньше медианы (2838.84) и среднего (3016.59). Положительный эксцесс (0.67) показал лептокуртическое распределение (островершинное) с тяжёлыми хвостами, что указывает на наличие экстремальных значений. Коэффициент вариации (26%) остаётся в пределах однородности совокупности, хотя разброс данных выше, чем в первой выборке.

Исправленные оценки дисперсии и СКО, как и ожидалось, оказались больше смещённых за счёт поправки на число степеней свободы. Для первой выборки исправленное СКО составило 14.78 против смещённого 14.72, для второй — 801.23 против 797.61. Различия между оценками более выражены во второй выборке в абсолютных значениях, но в относительном выражении сопоставимы с первой выборкой.